### PR TITLE
feat:[CDS-68821]: Consider Active app based on Service name if env not found

### PR DESCRIPTION
**Pull request by piyush-bhuwalka**
## Describe your changes

In case there is no app which contains ACTIVE env identifier, we need to consider the prefix named app as the prod app, so that it does not get deleted.

Additionally retry logic has been added when adding or removing env variables

Impact areas - 
1. For env variables there is no impact expected, as we only have added retry logic
2. For fetching the active application, previously it was just taking the last application from the list if none had the env. variable set, now we will match it by the release name as well and take that as the active application instead.
3. Refactoring of retry logic inside leaf nodes, and handling exceptions after that at necessary places

Test plan - https://docs.google.com/spreadsheets/d/1QGHBQKxX6oBKEOAy9tHmMRw17g0hXtDIXCR2Q4KJyGw/edit?usp=sharing

## Checklist
- [x] I've documented the changes in the PR description.
- [x] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>

You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/47699)

### CODE REVIEW

File: 930-delegate-tasks/src/main/java/io/harness/delegate/cf/PcfCommandTaskBaseHelper.java [hunk 1]
```diff
@@ -543,6 +544,21 @@ public ApplicationSummary findCurrentActiveApplication(List<ApplicationSummary>
...
+    if(!logPrinted) {
+      for (int i = previousReleases.size() - 1; i >= 0; i--) {
+        ApplicationSummary applicationSummary = previousReleases.get(i);
+
+        if (releaseNamePrefix.equals(applicationSummary.getName())) {
+          activeApplication = applicationSummary;
+          activeVersions.add(applicationSummary);
+          logPrinted = true;
+          executionLogCallback.saveExecutionLog(
+                  String.format("Found current Active App: [%s], as none of them have HARNESS__STATUS__IDENTIFIER set as ACTIVE",
```
Comment (piyush-bhuwalka) [inline 930-delegate-tasks/src/main/java/io/harness/delegate/cf/PcfCommandTaskBaseHelper.java:556]: change log msg

Comment (piyush-bhuwalka) [reply 930-delegate-tasks/src/main/java/io/harness/delegate/cf/PcfCommandTaskBaseHelper.java]: changed

**Comment by RishabhGupta34** [comment]
trigger uts

**Comment by RishabhGupta34** [comment]
trigger uts

**Comment by RishabhGupta34** [comment]
trigger messagecheck

**Review by anilchowdhury-harness — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/47699)* status: 15 file(s) to review. 1 change requests. 0 of 2 approvals obtained. 

___
*[`930-delegate-tasks/src/main/java/io/harness/delegate/cf/retry/RetryAbleTaskExecutorForEnvVariables.java` line 73 at r5](https://reviewable.io/reviews/harness/harness-core/47699#-NWNg9ng25YxMJKVynO0:-NWNg9nhBooRp9jQ4ff8:b4yich5) ([raw file](https://github.com/harness/harness-core/blob/18bc15e5fcf72771574ec98c2ac166e96de5c5c2/930-delegate-tasks/src/main/java/io/harness/delegate/cf/retry/RetryAbleTaskExecutorForEnvVariables.java#L73)):*
> ```Java
>     }
>   }
> }
> ```

Why we are creating new retryable executor. There is already one defined. We should use that one

**Review by anilchowdhury-harness — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/47699)* status: 15 file(s) to review. 1 change requests. 0 of 2 approvals obtained. 

___
*[`400-rest/src/main/java/software/wings/helpers/ext/pcf/CfDeploymentManagerUnsupported.java` line 219 at r5](https://reviewable.io/reviews/harness/harness-core/47699#-NWY7UQqBlf-rdrUqUdP:-NWY7UQqBlf-rdrUqUdQ:blhfy63) ([raw file](https://github.com/harness/harness-core/blob/18bc15e5fcf72771574ec98c2ac166e96de5c5c2/400-rest/src/main/java/software/wings/helpers/ext/pcf/CfDeploymentManagerUnsupported.java#L219)):*
> ```Java
>       LogCallback executionLogCallback) throws PivotalClientApiException {
>     throw new PivotalClientApiException("PCF operations not supported by this API.");
>   }
> ```

Why do we need to create separate method for CG/NG ?

**Comment by piyush-bhuwalka** [comment]
*[`930-delegate-tasks/src/main/java/io/harness/delegate/cf/retry/RetryAbleTaskExecutorForEnvVariables.java` line 73 at r5](https://reviewable.io/reviews/harness/harness-core/47699#-NWNg9ng25YxMJKVynO0:-NWa5l6A7dWPJ1lmm5h-:bxgu3d) ([raw file](https://github.com/harness/harness-core/blob/18bc15e5fcf72771574ec98c2ac166e96de5c5c2/930-delegate-tasks/src/main/java/io/harness/delegate/cf/retry/RetryAbleTaskExecutorForEnvVariables.java#L73)):*
<details><summary><i>Previously, anilchowdhury-harness (Anil Chowdhury) wrote…</i></summary><blockquote>

Why we are creating new retryable executor. There is already one defined. We should use that one
</blockquote></details>

That retryable executor does not take into account the result of another task that needs to run. We need another task inside it to check whether env variable is set or not. The previous exectuor was insufficient for that

**Comment by piyush-bhuwalka** [comment]
*[`400-rest/src/main/java/software/wings/helpers/ext/pcf/CfDeploymentManagerUnsupported.java` line 219 at r5](https://reviewable.io/reviews/harness/harness-core/47699#-NWY7UQqBlf-rdrUqUdP:-NWa6OxH49daaSV8o81I:bvtf0ys) ([raw file](https://github.com/harness/harness-core/blob/18bc15e5fcf72771574ec98c2ac166e96de5c5c2/400-rest/src/main/java/software/wings/helpers/ext/pcf/CfDeploymentManagerUnsupported.java#L219)):*
<details><summary><i>Previously, anilchowdhury-harness (Anil Chowdhury) wrote…</i></summary><blockquote>

Why do we need to create separate method for CG/NG ?
</blockquote></details>

because the env variable is different in CG and NG for stage apps. and there is some logic in CG where they have HARNESS__STAGE__IDENTIFIER and HARNESS__STAGE__INDENTIFIER  (seems like a typo which they have tried a workaround for) which they are maintaining. We did not want to touch it

**Review by RishabhGupta34 — APPROVED** [review]

**Review by anilchowdhury-harness — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/47699)* status: Some of the checks are too old. 15 file(s) to review. 1 change requests. 2. unresolved discussions. 1 of 2 approvals obtained. 

___
*[`930-delegate-tasks/src/main/java/io/harness/delegate/cf/PcfCommandTaskBaseHelper.java` line 556 at r5](https://reviewable.io/reviews/harness/harness-core/47699#-NWbxZ8o6GFHUyRuScRl:-NWbxZ8o6GFHUyRuScRm:b-esvuan) ([raw file](https://github.com/harness/harness-core/blob/18bc15e5fcf72771574ec98c2ac166e96de5c5c2/930-delegate-tasks/src/main/java/io/harness/delegate/cf/PcfCommandTaskBaseHelper.java#L556)):*
> ```Java
>           logPrinted = true;
>           executionLogCallback.saveExecutionLog(
>               String.format("Found current Active App: [%s], as it has same name as release name specified by user",
> ```

Lets refactor this method, currently the core logic is not clear. 
Active app is found only in these 2 cases:-
1. Either HARNESS__STATUS__IDENTIFIER is set
2. If above env is not set, then whichever application name is equal to releaseNamePredix, i.e. service Name

Lets refactor the code this way

**Review by anilchowdhury-harness — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/47699)* status: Some of the checks are too old. 15 file(s) to review. 1 change requests. 2. unresolved discussions. 1 of 2 approvals obtained. 

___
*[`930-delegate-tasks/src/main/java/io/harness/delegate/cf/PcfRouteUpdateCommandTaskHandler.java` line 439 at r5](https://reviewable.io/reviews/harness/harness-core/47699#-NWby5r14TzguufGDOl-:-NWby5r14TzguufGDOl0:b-e3kyk0) ([raw file](https://github.com/harness/harness-core/blob/18bc15e5fcf72771574ec98c2ac166e96de5c5c2/930-delegate-tasks/src/main/java/io/harness/delegate/cf/PcfRouteUpdateCommandTaskHandler.java#L439)):*
> ```Java
>     retryAbleTaskExecutor.execute(
>         ()
>             -> pcfDeploymentManager.unsetEnvironmentVariableForAppStatus(cfRequestConfig, executionLogCallback),
> ```

Lets move the retry able logic inside the pcfDeploymentManager.unsetEnvironmentVariableForAppStatus. This way all the consumer will benefit from it

**Review by anilchowdhury-harness — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/47699)* status: Some of the checks are too old. 15 file(s) to review. 1 change requests. 2. unresolved discussions. 1 of 2 approvals obtained. 

___
*[`930-delegate-tasks/src/main/java/io/harness/delegate/cf/PcfRouteUpdateCommandTaskHandler.java` line 509 at r5](https://reviewable.io/reviews/harness/harness-core/47699#-NWbyHbV4_wc92gJH4CG:-NWbyHbV4_wc92gJH4CH:b-bcuxs9) ([raw file](https://github.com/harness/harness-core/blob/18bc15e5fcf72771574ec98c2ac166e96de5c5c2/930-delegate-tasks/src/main/java/io/harness/delegate/cf/PcfRouteUpdateCommandTaskHandler.java#L509)):*
> ```Java
>     retryAbleTaskExecutor.execute(()
>                                       -> pcfDeploymentManager.setEnvironmentVariableForAppStatus(
>                                           cfRequestConfig, isActiveApplication, executionLogCallback),
> ```

Lets move the retry able logic inside the pcfDeploymentManager.setEnvironmentVariableForAppStatus. This way all the consumer will benefit from it

**Review by anilchowdhury-harness — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/47699)* status: Some of the checks are too old. 15 file(s) to review. 1 change requests. 1. unresolved discussions. 1 of 2 approvals obtained. 

___
*[`930-delegate-tasks/src/main/java/io/harness/delegate/cf/retry/RetryAbleTaskExecutorForEnvVariables.java` line 73 at r5](https://reviewable.io/reviews/harness/harness-core/47699#-NWNg9ng25YxMJKVynO0:-NWbyNkrEfnuqNfHHROx:b-k3kenn) ([raw file](https://github.com/harness/harness-core/blob/18bc15e5fcf72771574ec98c2ac166e96de5c5c2/930-delegate-tasks/src/main/java/io/harness/delegate/cf/retry/RetryAbleTaskExecutorForEnvVariables.java#L73)):*
<details><summary><i>Previously, piyush-bhuwalka wrote…</i></summary><blockquote>

That retryable executor does not take into account the result of another task that needs to run. We need another task inside it to check whether env variable is set or not. The previous exectuor was insufficient for that
</blockquote></details>

As discussed we need to write a method which will verify if the variable has been set. If not then throw exception. The current retryable framework will be able to handle it, hence we don't need separate one

**Review by anilchowdhury-harness — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/47699)* status: Some of the checks are too old. 15 file(s) to review. 1 change requests. 1. unresolved discussions. 1 of 2 approvals obtained. 

___
*[`930-delegate-tasks/src/main/java/io/harness/delegate/cf/retry/RetryAbleTaskForVerfication.java` line 14 at r5](https://reviewable.io/reviews/harness/harness-core/47699#-NWbybQ43xsvd9CFzfX_:-NWbybQ43xsvd9CFzfXa:b-nszhst) ([raw file](https://github.com/harness/harness-core/blob/18bc15e5fcf72771574ec98c2ac166e96de5c5c2/930-delegate-tasks/src/main/java/io/harness/delegate/cf/retry/RetryAbleTaskForVerfication.java#L14)):*
> ```Java
> public interface RetryAbleTaskForVerfication {
>   boolean execute() throws PivotalClientApiException;
> }
> ```

We don't need this class

**Review by anilchowdhury-harness — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/47699)* status: Some of the checks are too old. 15 file(s) to review. 1 change requests. 1. unresolved discussions. 1 of 2 approvals obtained. 

___
*[`930-delegate-tasks/src/main/java/io/harness/delegate/pcf/CfSwapRollbackCommandTaskHandlerNG.java` line 530 at r5](https://reviewable.io/reviews/harness/harness-core/47699#-NWbyfHPDZCnwrzql8mz:-NWbyfHPDZCnwrzql8n-:b-c07m8c) ([raw file](https://github.com/harness/harness-core/blob/18bc15e5fcf72771574ec98c2ac166e96de5c5c2/930-delegate-tasks/src/main/java/io/harness/delegate/pcf/CfSwapRollbackCommandTaskHandlerNG.java#L530)):*
> ```Java
>             -> cfDeploymentManager.unsetEnvironmentVariableForAppStatus(cfRequestConfig, executionLogCallback),
>         executionLogCallback, log, retryPolicy,
>         () -> cfDeploymentManager.checkUnsettingEnvironmentVariableForAppStatus(cfRequestConfig, executionLogCallback));
> ```

Lets move the retry able logic inside the pcfDeploymentManager.checkUnsettingEnvironmentVariableForAppStatus. This way all the consumer will benefit from it

**Review by anilchowdhury-harness — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/47699)* status: Some of the checks are too old. 15 file(s) to review. 1 change requests. 1. unresolved discussions. 1 of 2 approvals obtained. 

___
*[`930-delegate-tasks/src/main/java/io/harness/delegate/pcf/CfSwapRouteCommandTaskHandlerNG.java` line 425 at r5](https://reviewable.io/reviews/harness/harness-core/47699#-NWbyst75skaTiLc1_qB:-NWbyst75skaTiLc1_qC:b-2fiseo) ([raw file](https://github.com/harness/harness-core/blob/18bc15e5fcf72771574ec98c2ac166e96de5c5c2/930-delegate-tasks/src/main/java/io/harness/delegate/pcf/CfSwapRouteCommandTaskHandlerNG.java#L425)):*
> ```Java
> 
>     retryAbleTaskExecutor.execute(()
>                                       -> cfDeploymentManager.setEnvironmentVariableForAppStatusNG(
> ```

Lets move the retry able logic inside the pcfDeploymentManager.setEnvironmentVariableForAppStatusNG. This way all the consumer will benefit from it
___
*[`930-delegate-tasks/src/main/java/io/harness/delegate/pcf/CfSwapRouteCommandTaskHandlerNG.java` line 426 at r5](https://reviewable.io/reviews/harness/harness-core/47699#-NWbyrW2Dneu6wdkfn3w:-NWbyrW2Dneu6wdkfn3x:b-c07m8c) ([raw file](https://github.com/harness/harness-core/blob/18bc15e5fcf72771574ec98c2ac166e96de5c5c2/930-delegate-tasks/src/main/java/io/harness/delegate/pcf/CfSwapRouteCommandTaskHandlerNG.java#L426)):*
> ```Java
>     retryAbleTaskExecutor.execute(()
>                                       -> cfDeploymentManager.setEnvironmentVariableForAppStatusNG(
>                                           cfRequestConfig, isActiveApplication, executionLogCallback),
> ```

Lets move the retry able logic inside the pcfDeploymentManager.checkUnsettingEnvironmentVariableForAppStatus. This way all the consumer will benefit from it

**Comment by piyush-bhuwalka** [comment]
trigger uts

**Comment by piyush-bhuwalka** [comment]
trigger uts

**Comment by piyush-bhuwalka** [comment]
trigger uts

**Comment by piyush-bhuwalka** [comment]
trigger uts

**Comment by piyush-bhuwalka** [comment]
trigger uts

**Comment by piyush-bhuwalka** [comment]
trigger uts

**Comment by piyush-bhuwalka** [comment]
trigger uts

**Review by anilchowdhury-harness — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/47699)* status: 35 file(s) to review. 1 change requests. 1. unresolved discussions. 0 of 2 approvals obtained. Waiting on team(s): del-team. 

___
*[`930-delegate-tasks/src/main/java/io/harness/delegate/cf/PcfCommandTaskBaseHelper.java` line 550 at r12](https://reviewable.io/reviews/harness/harness-core/47699#-NWugbgr8tqyk3RynWYe:-NWugbgs8Tr1RFwbqaf-:b-tr2b0i) ([raw file](https://github.com/harness/harness-core/blob/c4ab86dd4e60b0ea3d28f8cd769c7622626b03ea/930-delegate-tasks/src/main/java/io/harness/delegate/cf/PcfCommandTaskBaseHelper.java#L550)):*
> ```Java
>       throw new InvalidPcfStateException(msgBuilder.toString(), INVALID_INFRA_STATE, USER_SRE);
>     }
> ```

As discussed with you earlier , please move this validation inside findActiveBasedOnEnvironmentVariable()

**Comment by piyush-bhuwalka** [comment]
trigger uts

**Review by anilchowdhury-harness — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/47699)* status: 35 file(s) to review. 1. unresolved discussions. 0 of 2 approvals obtained. Waiting on team(s): del-team. 

___
*[`930-delegate-tasks/src/main/java/io/harness/delegate/cf/PcfCommandTaskBaseHelper.java` line 580 at r13](https://reviewable.io/reviews/harness/harness-core/47699#-NWupEvw-2nkWKOml36W:-NWupEvw-2nkWKOml36X:bkjaedl) ([raw file](https://github.com/harness/harness-core/blob/4721099879d3f043f74d3cd9be819905fe75b8be/930-delegate-tasks/src/main/java/io/harness/delegate/cf/PcfCommandTaskBaseHelper.java#L580)):*
> ```Java
>     }
> 
>     return activeApplication;
> ```

why do we need logPrinted flag now ?

**Comment by piyush-bhuwalka** [comment]
*[`930-delegate-tasks/src/main/java/io/harness/delegate/cf/PcfCommandTaskBaseHelper.java` line 580 at r13](https://reviewable.io/reviews/harness/harness-core/47699#-NWupEvw-2nkWKOml36W:-NWurdoA-qKnkCmw56Ez:bq2rasn) ([raw file](https://github.com/harness/harness-core/blob/4721099879d3f043f74d3cd9be819905fe75b8be/930-delegate-tasks/src/main/java/io/harness/delegate/cf/PcfCommandTaskBaseHelper.java#L580)):*
<details><summary><i>Previously, anilchowdhury-harness (Anil Chowdhury) wrote…</i></summary><blockquote>

why do we need logPrinted flag now ?
</blockquote></details>

Its removed in much earlier commit . Its not there

**Comment by piyush-bhuwalka** [comment]
*[`930-delegate-tasks/src/main/java/io/harness/delegate/cf/PcfCommandTaskBaseHelper.java` line 556 at r5](https://reviewable.io/reviews/harness/harness-core/47699#-NWbxZ8o6GFHUyRuScRl:-NWurmhTA9q7Va9oCZcl:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/18bc15e5fcf72771574ec98c2ac166e96de5c5c2/930-delegate-tasks/src/main/java/io/harness/delegate/cf/PcfCommandTaskBaseHelper.java#L556)):*
<details><summary><i>Previously, anilchowdhury-harness (Anil Chowdhury) wrote…</i></summary><blockquote>

Lets refactor this method, currently the core logic is not clear. 
Active app is found only in these 2 cases:-
1. Either HARNESS__STATUS__IDENTIFIER is set
2. If above env is not set, then whichever application name is equal to releaseNamePredix, i.e. service Name

Lets refactor the code this way
</blockquote></details>

Done.

**Comment by piyush-bhuwalka** [comment]
*[`930-delegate-tasks/src/main/java/io/harness/delegate/cf/PcfRouteUpdateCommandTaskHandler.java` line 509 at r5](https://reviewable.io/reviews/harness/harness-core/47699#-NWbyHbV4_wc92gJH4CG:-NWurosq7at-yJqV8eXB:bb6un1j) ([raw file](https://github.com/harness/harness-core/blob/18bc15e5fcf72771574ec98c2ac166e96de5c5c2/930-delegate-tasks/src/main/java/io/harness/delegate/cf/PcfRouteUpdateCommandTaskHandler.java#L509)):*
<details><summary><i>Previously, anilchowdhury-harness (Anil Chowdhury) wrote…</i></summary><blockquote>

Lets move the retry able logic inside the pcfDeploymentManager.setEnvironmentVariableForAppStatus. This way all the consumer will benefit from it
</blockquote></details>

Done

**Comment by piyush-bhuwalka** [comment]
*[`930-delegate-tasks/src/main/java/io/harness/delegate/cf/PcfRouteUpdateCommandTaskHandler.java` line 439 at r5](https://reviewable.io/reviews/harness/harness-core/47699#-NWby5r14TzguufGDOl-:-NWurqbjCSiSZ1ByROGb:bb6un1j) ([raw file](https://github.com/harness/harness-core/blob/18bc15e5fcf72771574ec98c2ac166e96de5c5c2/930-delegate-tasks/src/main/java/io/harness/delegate/cf/PcfRouteUpdateCommandTaskHandler.java#L439)):*
<details><summary><i>Previously, anilchowdhury-harness (Anil Chowdhury) wrote…</i></summary><blockquote>

Lets move the retry able logic inside the pcfDeploymentManager.unsetEnvironmentVariableForAppStatus. This way all the consumer will benefit from it
</blockquote></details>

Done

**Comment by piyush-bhuwalka** [comment]
*[`930-delegate-tasks/src/main/java/io/harness/delegate/pcf/CfSwapRouteCommandTaskHandlerNG.java` line 425 at r5](https://reviewable.io/reviews/harness/harness-core/47699#-NWbyst75skaTiLc1_qB:-NWurs5L-yg_NW_I8lkm:bb6un1j) ([raw file](https://github.com/harness/harness-core/blob/18bc15e5fcf72771574ec98c2ac166e96de5c5c2/930-delegate-tasks/src/main/java/io/harness/delegate/pcf/CfSwapRouteCommandTaskHandlerNG.java#L425)):*
<details><summary><i>Previously, anilchowdhury-harness (Anil Chowdhury) wrote…</i></summary><blockquote>

Lets move the retry able logic inside the pcfDeploymentManager.setEnvironmentVariableForAppStatusNG. This way all the consumer will benefit from it
</blockquote></details>

Done

**Comment by piyush-bhuwalka** [comment]
*[`930-delegate-tasks/src/main/java/io/harness/delegate/pcf/CfSwapRouteCommandTaskHandlerNG.java` line 426 at r5](https://reviewable.io/reviews/harness/harness-core/47699#-NWbyrW2Dneu6wdkfn3w:-NWursol7_A6srR2b0Fj:bb6un1j) ([raw file](https://github.com/harness/harness-core/blob/18bc15e5fcf72771574ec98c2ac166e96de5c5c2/930-delegate-tasks/src/main/java/io/harness/delegate/pcf/CfSwapRouteCommandTaskHandlerNG.java#L426)):*
<details><summary><i>Previously, anilchowdhury-harness (Anil Chowdhury) wrote…</i></summary><blockquote>

Lets move the retry able logic inside the pcfDeploymentManager.checkUnsettingEnvironmentVariableForAppStatus. This way all the consumer will benefit from it
</blockquote></details>

Done

**Comment by piyush-bhuwalka** [comment]
*[`930-delegate-tasks/src/main/java/io/harness/delegate/cf/retry/RetryAbleTaskForVerfication.java` line 14 at r5](https://reviewable.io/reviews/harness/harness-core/47699#-NWbybQ43xsvd9CFzfX_:-NWurtsj3IN9YZ5t4wmU:bb6un1j) ([raw file](https://github.com/harness/harness-core/blob/18bc15e5fcf72771574ec98c2ac166e96de5c5c2/930-delegate-tasks/src/main/java/io/harness/delegate/cf/retry/RetryAbleTaskForVerfication.java#L14)):*
<details><summary><i>Previously, anilchowdhury-harness (Anil Chowdhury) wrote…</i></summary><blockquote>

We don't need this class
</blockquote></details>

Done

**Comment by piyush-bhuwalka** [comment]
*[`930-delegate-tasks/src/main/java/io/harness/delegate/cf/retry/RetryAbleTaskExecutorForEnvVariables.java` line 73 at r5](https://reviewable.io/reviews/harness/harness-core/47699#-NWNg9ng25YxMJKVynO0:-NWururm1XjtWGlTvqUJ:bb6un1j) ([raw file](https://github.com/harness/harness-core/blob/18bc15e5fcf72771574ec98c2ac166e96de5c5c2/930-delegate-tasks/src/main/java/io/harness/delegate/cf/retry/RetryAbleTaskExecutorForEnvVariables.java#L73)):*
<details><summary><i>Previously, anilchowdhury-harness (Anil Chowdhury) wrote…</i></summary><blockquote>

As discussed we need to write a method which will verify if the variable has been set. If not then throw exception. The current retryable framework will be able to handle it, hence we don't need separate one
</blockquote></details>

Done

**Comment by piyush-bhuwalka** [comment]
*[`930-delegate-tasks/src/main/java/io/harness/delegate/pcf/CfSwapRollbackCommandTaskHandlerNG.java` line 530 at r5](https://reviewable.io/reviews/harness/harness-core/47699#-NWbyfHPDZCnwrzql8mz:-NWurwBj96qTEUqExciO:bb6un1j) ([raw file](https://github.com/harness/harness-core/blob/18bc15e5fcf72771574ec98c2ac166e96de5c5c2/930-delegate-tasks/src/main/java/io/harness/delegate/pcf/CfSwapRollbackCommandTaskHandlerNG.java#L530)):*
<details><summary><i>Previously, anilchowdhury-harness (Anil Chowdhury) wrote…</i></summary><blockquote>

Lets move the retry able logic inside the pcfDeploymentManager.checkUnsettingEnvironmentVariableForAppStatus. This way all the consumer will benefit from it
</blockquote></details>

Done

**Comment by piyush-bhuwalka** [comment]
*[`930-delegate-tasks/src/main/java/io/harness/delegate/cf/PcfCommandTaskBaseHelper.java` line 550 at r12](https://reviewable.io/reviews/harness/harness-core/47699#-NWugbgr8tqyk3RynWYe:-NWurxxU3VamZDztA3Fo:bb6un1j) ([raw file](https://github.com/harness/harness-core/blob/c4ab86dd4e60b0ea3d28f8cd769c7622626b03ea/930-delegate-tasks/src/main/java/io/harness/delegate/cf/PcfCommandTaskBaseHelper.java#L550)):*
<details><summary><i>Previously, anilchowdhury-harness (Anil Chowdhury) wrote…</i></summary><blockquote>

As discussed with you earlier , please move this validation inside findActiveBasedOnEnvironmentVariable()
</blockquote></details>

Done

**Review by anilchowdhury-harness — COMMENTED** [review]
:lgtm:

> Reviewed 1 of 7 files at r1, 5 of 13 files at r6, 14 of 24 files at r7, 4 of 7 files at r8, 1 of 1 files at r9, 6 of 9 files at r10, 3 of 3 files at r12, 1 of 1 files at r13, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/47699#-:-NWusPpM3JjCQFIU7vuT:bnfp4nl)* status: 35 file(s) to review. 11. unresolved discussions. 0 of 2 approvals obtained. Waiting on team(s): del-team.

**Comment by piyush-bhuwalka** [comment]
trigger commonchecks

**Comment by piyush-bhuwalka** [comment]
trigger codebasehash

**Comment by piyush-bhuwalka** [comment]
trigger smartchecks

**Comment by piyush-bhuwalka** [comment]
trigger pmd

**Review by anilchowdhury-harness — APPROVED** [review]
:lgtm:

> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/47699#-:-NWvyNmL9fec6Bu-phIY:bnfp4nl)* status: 35 file(s) to review. 11. unresolved discussions. 1 of 2 approvals obtained. Waiting on team(s): del-team.

**Review by RishabhGupta34 — APPROVED** [review]

**Comment by piyush-bhuwalka** [comment]
trigger review

**Review by anilchowdhury-harness — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/47699)* status: 35 file(s) to review. Waiting on team(s): del-team.

**Pull request merged**